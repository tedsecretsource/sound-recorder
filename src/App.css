body {
  display: flex;
  flex-direction: column;
  height: 100vh;
  margin: 0;
  background-color: rgb(27, 30, 38);
  color: rgb(0,162,144)
}

a {
  color: rgb(0,162,144);
}

a:visited {
  color: rgb(0,206,186);
}

a.logo {
  text-decoration: none;
  display: flex;
  flex-direction: column;
  align-items: center;
}

a.logo div {
  font-size: 1rem;
}

img.logo-file {
  height: 100px;
  width: 100px;
  margin: 10px;
}

div#root {
  display: flex;
  flex-direction: column;
  flex-grow: 1;
  height: 100vh;
  overflow-y: auto;
}

header {
  position: sticky;
  top: 0;
  display: flex;
  flex-direction: row;
  justify-content: center;
  background-color: black;
}

main {
  flex-grow: 1;
  overflow-y: auto;
  padding: 20px;
  display: flex;
  flex-direction: column;
}

footer {
  position: sticky;
  bottom: 0;
}


/* https://stackoverflow.com/questions/1575141/how-to-make-a-div-100-height-of-the-browser-window */
/* div#root {
  margin: 0.5rem;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: flex-start;
  min-height: 100vh;
  place-content: space-between;
}

main {
  width: 100%;
  justify-content: center;
  display: flex;
  flex-direction: column;
}

header {
  display: flex;
  width: 100%;
  justify-content: center;
} */
