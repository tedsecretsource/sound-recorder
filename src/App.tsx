import { useState, useEffect } from 'react'
import { Link, Outlet, useLocation } from "react-router-dom"
import Footer from './components/Footer'
import './App.css';
import Logo from './components/Logo'
import Recorder from './components/Recorder';

function App() {

  let location = useLocation()

  const [outlet, setOutlet] = useState(<Outlet />)

  useEffect(() => {
    if( location.pathname === '/' ) {
      setOutlet(<Recorder />)
    } else {
      setOutlet(<Outlet />)
    }
  }, [location])

  return (
    <>
    <header>
      <h1 style={{marginTop: "0"}}>
        <Link to="/" className='logo'>
          <Logo />
          <div>Sound Recorder</div>
        </Link>
      </h1>
    </header>
    <main>
      {outlet}
    </main>
    <Footer />
    </>
  );
}

export default App;
