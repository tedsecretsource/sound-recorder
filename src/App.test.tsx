import renderer from 'react-test-renderer';
import {render, screen} from '@testing-library/react';
import App from './App';


const setupMockedMediaDevices = () => {
  const mockMediaDevices = {
    getUserMedia: jest.fn().mockResolvedValueOnce('fake data' as any),
  }

  Object.defineProperty(window.navigator, 'mediaDevices', {
    writable: true,
    value: mockMediaDevices,
  })

  window.MediaStream = jest.fn().mockImplementation(() => ({
    addTrack: jest.fn()
  }))
}


describe('With an empty list of recordings', () => {
  beforeEach(() => {
    setupMockedMediaDevices()
  })

  test('renders Sound Recorder title', async () => {
    const component = await renderer.create(
      <App />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('has link to license, terms, and copyright notice', async () => {
    render(<App />)
    expect(screen.getByText('Terms of Use')).toBeInTheDocument()
    expect(screen.getByText('Terms of Use')).toHaveAttribute('href', './terms_of_use')
    expect(screen.getByText('License')).toBeInTheDocument()
    expect(screen.getByText('License')).toHaveAttribute('href', 'https://github.com/tedsecretsource/sound-recorder/blob/main/LICENSE.md')
    expect(screen.getByText('Â© Copyright Secret Source Technology 2022')).toBeInTheDocument()
  });
})