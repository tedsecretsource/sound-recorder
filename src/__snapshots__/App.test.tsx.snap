// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With an empty list of recordings renders Sound Recorder title 1`] = `
<DocumentFragment>
  <header>
    <h1>
      <a
        href="#/"
      >
        Sound Recorder
      </a>
    </h1>
  </header>
  <main />
  <footer>
    <p>
      Â© Copyright Secret Source Technology 2022
    </p>
    <p>
      <a
        href="#/terms-and-conditions"
      >
        Terms & Conditions
      </a>
    </p>
    <p>
      <a
        href="https://github.com/tedsecretsource/sound-recorder/blob/main/LICENSE.md"
        rel="noreferrer"
        target="_blank"
      >
        License
      </a>
    </p>
  </footer>
</DocumentFragment>
`;
